x/subscription/keeper/cu_tracker.go

```diff
@@ -228,26 +230,41 @@ func (k Keeper) RewardAndResetCuTracker(ctx sdk.Context, cuTrackerTimerKeyBytes
 			)
 		} else {
 			utils.LogLavaEvent(ctx, k.Logger(ctx), types.MonthlyCuTrackerProviderRewardEventName, map[string]string{
-				"provider":       provider,
-				"sub":            sub,
-				"plan":           plan.Index,
-				"tracked_cu":     strconv.FormatUint(trackedCu, 10),
-				"plan_price":     plan.Price.String(),
-				"reward":         providerReward.String(),
-				"block":          strconv.FormatInt(ctx.BlockHeight(), 10),
-				"adjustment_raw": providerAdjustment.String(),
+				"provider":         provider,
+				"sub":              sub,
+				"plan":             subObj.PlanIndex,
+				"tracked_cu":       strconv.FormatUint(trackedCu, 10),
+				"credit_used":      creditToSub.String(),
+				"credit_remaining": subObj.Credit.String(),
+				"reward":           providerReward.String(),
+				"block":            strconv.FormatInt(ctx.BlockHeight(), 10),
+				"adjustment_raw":   providerAdjustment.String(),
 			}, "Provider got monthly reward successfully")
 		}
 	}
 
+	var latestSub types.Subscription
+	latestEntryBlock, _, _, found := k.subsFS.FindEntryDetailed(ctx, subObj.Consumer, uint64(ctx.BlockHeight()), &latestSub)
+	if found {
+		latestSub.Credit = latestSub.Credit.SubAmount(totalTokenAmount)
+		k.subsFS.ModifyEntry(ctx, latestSub.Consumer, latestEntryBlock, &latestSub)
+	}
+
 	// send remainder of rewards to the community pool
 	rewardsRemainder := totalTokenAmount.Sub(totalTokenRewarded)
 	if !rewardsRemainder.IsZero() {
-		err = k.rewardsKeeper.FundCommunityPoolFromModule(ctx, rewardsRemainder, types.ModuleName)
-		if err != nil {
-			utils.LavaFormatError("failed sending remainder of rewards to the community pool", err,
-				utils.Attribute{Key: "rewards_remainder", Value: rewardsRemainder.String()},
-			)
+		if found {
+			// sub not expired yet, return rewards remainder to credit
+			latestSub.Credit = latestSub.Credit.AddAmount(rewardsRemainder)
```

Comment:
why do we do that again?
do it one time at the end
what you do:
* credit = credit - totalTokenAmount
* credit = credit - rewardsRemainder

BUT
totalTokenAmount includes rewardsRemainder, so this is a bug
(look at rewardsRemainder := totalTokenAmount.Sub(totalTokenRewarded))